File: algorithms/400-499/435. Non-overlapping Intervals.cpp
class Solution {
public:
	int eraseOverlapIntervals(vector<vector<int>>& intervals) {
		if (intervals.empty()){
			return 0;
		}
		sort(intervals.begin(), intervals.end(), [](vector<int> a, vector<int> b){
			return a[1] < b[1];
		});
		int result = 0;
		int prev = intervals[0][1];
		for (int i = 1; i < intervals.size(); ++i){
			if (intervals[i][0] < prev){
				result += 1;
			}else{
				prev = intervals[i][1];
			}
		}
		return result;
	}
};

File: algorithms/400-499/435. Non-overlapping Intervals.py
class Solution:
	def eraseOverlapIntervals(self, intervals: List[List[int]]) -> int:
		if not intervals:
			return 0
		intervals.sort(key = lambda interval : interval[1])
		res = 0
		prev = intervals[0][1]
		for i in range(1, len(intervals)):
			if intervals[i][0] < prev:
				res +=1 
			else:
				prev = intervals[i][1]
		return res
			